tools/autograd/templates/VariableType.cpp

```diff
@@ -200,16 +200,18 @@ static void wrap_output(std::tuple<Variable, Variable, Variable>& t, FunctionFla
 }
 
 static void increment_version(const Tensor & t) {
-  auto pImpl = static_cast<VariableImpl*>(t.get());
-  (*pImpl->version_counter)++;
+  // accept a const Tensor & because some functions (e.g. relu) mark inputs as
+  // const even though they may be modified in-place.
```

Comment:
The const correctness story of ATen is a disaster (stemming from the disaster that is C++ const correctness, see https://github.com/zdevito/ATen/issues/27) but the consistent approach would be to make `version_counter()` in `Variable` self-const. I think you'll find that will compile.